core/src/main/java/discord4j/core/object/entity/channel/ForumChannel.java

```diff
@@ -251,18 +152,142 @@ public Mono<ThreadChannel> startThread(StartThreadInForumChannelSpec request) {
     }
 
     /**
-     * Requests to edit the current forum channel object
+     * Starts a new {@link ThreadChannel} in this forum channel. Properties specifying how to create this channel
+     * can be set via the {@code withXxx} methods of the returned {@link StartThreadInForumChannelMono}.
      *
-     * @param spec an immutable object that specifies the modifications requested
-     * @return A {@link Mono} that, upon completion, emits the updated {@link ForumChannel} object
+     * @param name The name of the thread
+     * @param messageCreateSpec The message to start the thread with
+     * @return A {@link StartThreadInForumChannelMono} where, upon successful completion, emits the created
+     * {@link ThreadChannel}. If an error is received, it is emitted through the {@code Mono}.
      */
-    public Mono<ForumChannel> edit(ForumChannelEditSpec spec) {
-        Objects.requireNonNull(spec);
-        return Mono.defer(
-                () -> getClient().getRestClient().getChannelService()
-                    .modifyChannel(getId().asLong(), spec.asRequest(), spec.reason()))
-            .map(data -> EntityUtil.getChannel(getClient(), data))
-            .cast(ForumChannel.class);
+    public StartThreadInForumChannelMono startThread(String name, ForumThreadMessageCreateSpec messageCreateSpec) {
+        return StartThreadInForumChannelMono.of(name, messageCreateSpec, this);
+    }
+
+    /**
+     * Request to retrieve all threads in this channel.
+     *
+     * @return A {@link Flux} that continually emits the {@link ThreadChannel threads} of the channel. If an error is
+     * received, it is emitted through the {@code Flux}.
+     */
+    public Flux<ThreadChannel> getAllThreads() {
+        return getClient().getGuildChannels(getGuildId())
+            .ofType(ThreadChannel.class)
```

Comment:
This does not work :(

Reply:
how?

Reply:
I assumed that threads were returned by the "/guilds/id/channels" endpoint, but no, unfortunately